geps/gep-91.md

```diff
@@ -16,7 +16,71 @@ This GEP proposes a way to validate the TLS certificate presented by the downstr
 ## Non-Goals
 - Define other fields that can be used to verify the client certificate such as the Cerificate Hash or Subject Alt Name. 
 
+### API
+
+* Introduce a `clientValidation` field with [Gateway.Tls][] that can be used to validate the client intiating the TLS connection
+to the Gateway
+* Introduce a `caCerficateRefs` field within `clientValidation` that can be used to specify a list of CA Certificates that
+can be used as a trusted anchor to validate the certificates presented by the client
+
+#### GO
+
+```go
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
+    // CACertificateRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates of
+    // the Certificate Authorities that can be used to
+    // as a trusted anchor to validate the certificates presented by the client.
+    //
+    // A single CACertificateRef to a Kubernetes ConfigMap with a key called `ca.crt`
+    // has "Core" support.
+    // Implementations MAY choose to support attaching multiple certificates to
+    // a Listener, but this behavior is implementation-specific.
+    //
+    // References to a resource in different namespace are invalid UNLESS there
+    // is a ReferenceGrant in the target namespace that allows the certificate
+    // to be attached. If a ReferenceGrant does not allow this reference, the
+    // "ResolvedRefs" condition MUST be set to False for this listener with the
+    // "RefNotPermitted" reason.
+    //
+    // +kubebuilder:validation:MaxItems=64
+    // +optional
+    CACertificateRefs []corev1.ObjectReference `json:”caCertificateRefs,omitempty”`
+}
+
+```
+
+#### YAML
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: mtls-basic
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: foo-https
+    protocol: HTTPS
+    port: 443
+    hostname: foo.example.com
+    tls:
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: foo-example-com-cert
+      clientValidation:
+        caCertificateRefs:
+        - kind: ConfigMap
+          group: ""
+          name: foo-example-com-ca-cert
```

Comment:
I think Istio users would like to be able to point this to a Secret (instead of ConfigMap), specifically the same secret that `certificateRefs` point to, only with an additional `ca.crt` key containing the client's CA cert.

@howardjohn WDYT?

Reply:
Yes, secret reference is common practice in CNCF landscape, pioneered by cert-manager. Not in the core `tls` SEcret type, but its de-facto standard.

I don't mind config map as well of course.

Reply:
This issue is still live? 

In istio I can do that with:
```yaml
    tls:
      mode: Terminate
      certificateRefs:
      - name: httpbin-credential
      options:
        gateway.istio.io/tls-terminate-mode: MUTUAL
```

I try to setup gateway api with cilium (1.15.5) withtout istio and other sidecars, but MTLS not working
```yaml
    tls:
      certificateRefs:
      - kind: Secret
        name: demo-cert
      frontendValidation:
        caCertificateRefs:
        - kind: Secret
          group: ""
          name: demo-cert
```